src/test/java/org/jenkinsci/plugins/workflow/steps/CoreWrapperStepTest.java

```diff
@@ -286,4 +287,109 @@ public static Slave createSpecialEnvSlave(JenkinsRule rule, String nodeName, @Ch
         }
     }
 
+    public static final class WrapperWithWorkspaceRequirement extends SimpleBuildWrapper {
+        @DataBoundConstructor public WrapperWithWorkspaceRequirement() { }
+        @Override public void setUp(@Nonnull Context context, @Nonnull Run<?, ?> build, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull TaskListener listener, @Nonnull EnvVars initialEnvironment) throws IOException, InterruptedException {
+            listener.getLogger().println(">>> workspace context required and provided.");
+            context.setDisposer(new DisposerWithWorkspaceRequirement());
+        }
+        // TODO: Mark as @Override once the minimum core version for this plugin is 2.258 or newer.
+        public void setUp(@Nonnull Context context, @Nonnull Run<?, ?> build, @Nonnull TaskListener listener, @Nonnull EnvVars initialEnvironment) throws IOException, InterruptedException {
+            listener.getLogger().println(">>> workspace context required but not provided!");
+            context.setDisposer(new DisposerWithWorkspaceRequirement());
+        }
+        public static final class DisposerWithWorkspaceRequirement extends Disposer {
+            @Override public void tearDown(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+                listener.getLogger().println("<<< workspace context required and provided.");
+            }
+            // TODO: Mark as @Override once the minimum core version for this plugin is 2.258 or newer.
+            public void tearDown(@Nonnull Run<?, ?> build, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+                listener.getLogger().println("<<< workspace context required but not provided!");
+            }
+        }
+        @Symbol("wrapperWithWorkspaceRequirement")
+        @TestExtension("wrapperWithWorkspaceRequirement")
+        public static class DescriptorImpl extends BuildWrapperDescriptor {
+            @Override public boolean isApplicable(AbstractProject<?, ?> project) { return true; }
+        }
+    }
+
+    @Issue("JENKINS-46175")
+    @Test
+    public void wrapperWithWorkspaceRequirement() throws Exception {
+        story.addStep(new Statement() {
```

Comment:
Prefer `story.then((JenkinsRule foo) -> { ... })` instead in new code. I'll file a PR to mark this method as deprecated in `jenkins-test-harness`.

Reply:
See https://github.com/jenkinsci/jenkins-test-harness/pull/244.